### PR TITLE
:seedling:  Update golangci-lint action to v3

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <kmuldoon@vmware.com>

Note this is a test and should not be merged or reviewed.

### CODE REVIEW

File: .github/workflows/golangci-lint.yml
Comment (killianmuldoon) [inline .github/workflows/golangci-lint.yml:19]: This had to be added v2 was able to install its own go env, but v3 requires this explicitly.

**Comment by sbueringer** [comment]
Looks good. I guess this would trigger a rebase of the other PR with the failed GitHub action which we wanted to use for the Prow issue.

Can you maybe open another PR to reproduce `/retest doesn't work`? I think even a simple change like changing the name of the `generate-module` target should be enough to reproduce the Prow issue. WDYT?

/lgtm

**Comment by killianmuldoon** [comment]
DO NOT MERGE

**Comment by killianmuldoon** [comment]
Sure - this PR breaks our logic for getting the golanglint-ci version though :| so `make lint` seems to fail. Luckily enough I caught it before merge.

**Comment by killianmuldoon** [comment]
I added a minimal fix to the Makefile to ensure we pull out the correct version. The logic is just as brittle as before, however, though it's currently correct. I'll create a help-wanted issue to make it more resilient if we're alright with using this for now.

**Comment by sbueringer** [comment]
@killianmuldoon fine for me as is. I think we change the GitHub action so rarely that it's not necessary to improve the bash snippet. 

Even if we change it, we use make lint so regularly that it wouldn't be broken for long.

/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**
